src/prometheus_server.py

```diff
@@ -1,17 +1,36 @@
-import json
-import urllib3
+import logging
+from requests import get, post
+from requests.exceptions import ConnectionError, ConnectTimeout
+
+logger = logging.getLogger(__name__)
 
 
 class Prometheus:
-    def __init__(self, host, port):
+    def __init__(self, host, port, api_timeout=2.0):
         """Utility to manage a Prometheus application.
         Args:
             host: host address of Prometheus application.
             port: port on which Prometheus service is exposed.
         """
-        self.host = host
-        self.port = port
-        self.http = urllib3.PoolManager()
+        self.base_url = "http://{}:{}".format(host, port)
+        self.api_timeout = api_timeout
+
+    def trigger_configuration_reload(self):
```

Comment:
Minor: rather wordy method name. If context (name of object this method is used on) is considered a shorter method name may suffice. For example `self._prometheus.reload_configuration()` .